android/capacitor/src/main/java/com/getcapacitor/plugin/util/CapacitorHttpUrlConnection.java

```diff
@@ -224,7 +225,16 @@ public void setRequestBody(PluginCall call, JSValue body, String bodyType) throw
                 this.writeRequestBody(body.toString());
             }
         } else if (bodyType != null && bodyType.equals("formData")) {
-            this.writeFormDataRequestBody(contentType, body.toJSArray());
+            String boundary = extractBoundaryFromContentType(contentType);
+            if (boundary == null) {
+                // If no boundary is provided, generate a random one and set the Content-Type header accordingly
+                // or otherwise servers will not be able to parse the request body. Browsers do this automatically
+                // but here we need to do this manually in order to comply with browser api behavior.
+                boundary = UUID.randomUUID().toString();
```

Comment:
Note that this part was missing in the iOS implementation but present for Android already.